### ISSUE TITLE
Sync IaC resources deployed by DataAll & manual deletion of objects from AWS Console

### ISSUE THREAD

**Issue by MATTT-P** [awslabs/aws-dataall#310]
### Describe the bug

DataAll is out of sync when user carry out activities within AWS management console, such as delete bucket or folder, the deleted bucket or folder is still appearing in DataAll . Also, user in Datahub still be able to create folder within the deleted bucket.

When dataset is deleted in DA, the associated S3 bucket in AWS management console will need to be deleted manually by user.

### How to Reproduce

Delete a bucket in AWS console.
You can still access and create folders within the deleted bucket from DataAll

### OS

Win

### Python version

-

### AWS data.all version

0.9

**Comment by dlpzx** [comment]
Hi @MATTT-P,

Here there are 3 different topics:

1. Bug on creation of folders in deleted buckets. I have checked the code and it is not possible that this happens since the boto3 call of creating an S3 prefix would throw an error. I also tested it directly: I deleted the S3 Bucket of a dataset and I tried to create a folder in that dataset. I got the following error: " "An error occurred (NoSuchBucket) when calling the PutObject operation: The specified bucket does not exist"," Try again to see if the error persists or if you are referring to another error.

2. Sync deleted S3 Bucket. When using Infrastructure as Code to deploy resources (as the Dataset CloudFormation stack) we discourage users to manually edit or edit the resources of the stack. This is not good practice and it removes the advantages of working with IaC, that is why we do not expect users to directly delete the S3 Bucket.

3. Sync of deleted folders. Yes, this is something that we have thought about. For example in a similar way that there is a syncing tables ECS task that runs on schedule we could implement a sync folders tasks that runs at a defined frequency.  Would this be sufficient for your organisation? Are you thinking of other ways of implementing this?

Bests
